app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt

```diff
@@ -250,4 +327,29 @@ class BluetoothSensorManager : SensorManager {
             )
         )
     }
+
+    fun updateBeaconMonitoringSensor(context: Context) {
+        val icon = if (MonitoringManager.isMonitoring()) "mdi:bluetooth" else "mdi:bluetooth-off"
+
+        var state = if (!isBtOn(context)) "Bluetooth is turned off" else if (MonitoringManager.isMonitoring()) "Monitoring" else "Stopped"
```

Comment:
So I noticed that on a fresh install on android 12 if I dont enable the BT state sensor the state of this sensor remains at "Bluetooth is turned off"  eventhough BT is on for this device. As soon as I enable BT state sensor the monitor starts reporting beacons and the state updates too. I think we may need to change the check on `isBtOn()` to not be sensor specific like it is today. The BLE transmit sensor does not have this particular issue. Or we may just need to add more BT permissions for Android 12 devices to work around it.

Reply:
I changed it to the `BluetoothUtils.isOn(context)` function which seem to fix this case. However when I switch Bluetooth of in the emulator on Android 12 the sensor state keeps showing `Monitoring`. I'm not sure if this happens on real devices too. Could you please check this?

Reply:
latest debug APK works well on my pixel 6 pro :)  the state also updates immediately when I toggle BT on/off